state-synchronizer/src/executor_proxy.rs

```diff
@@ -195,6 +198,25 @@ impl ExecutorProxyTrait for ExecutorProxy {
         self.storage.get_epoch_ending_ledger_info(version)
     }
 
+    /// Returns the ledger's timestamp for the given version in microseconds
+    fn get_vm_timestamp_for_version(&self, version: u64) -> Result<u64> {
+        self.storage
+            .get_account_state_with_proof_by_version(
```

Comment:
Probably more efficient and reliable to do `storage.get_block_timestamp(version)`

Reply:
oh god... nice